test/core/symbolFactoriesTests.ts

```diff
@@ -0,0 +1,103 @@
+///<reference path="../testReference.ts" />
+
+describe("SymbolFactory", () => {
+  describe("SymbolFactory generates correct path", () => {
+    let svg: d3.Selection<void>;
+    let symbolSize = 5;
+
+    beforeEach(() => {
+      svg = TestMethods.generateSVG();
+    });
+
+    it("returns a circle factory that generates circle based on symbolSize", () => {
+      let circleFactory = Plottable.SymbolFactories.circle();
+      let actualSize = Math.PI * Math.pow(symbolSize / 2, 2);
+      let d = circleFactory(symbolSize);
+      let expectedD = d3.svg.symbol().type("circle").size(actualSize)(null);
+      assert.strictEqual(d, expectedD, "a circle of set size is generated");
+      svg.append("path").attr("d", d);
+      let bbox = Plottable.Utils.DOM.elementBBox(svg.select("path"));
+      assert.strictEqual(bbox.height, symbolSize, "height is as set");
+      assert.strictEqual(bbox.width, symbolSize, "width is as set");
+      assert.strictEqual(bbox.x, -bbox.width / 2, "x is centered in the middle");
+      assert.strictEqual(bbox.y, -bbox.height / 2, "y is centered in the middle");
+      svg.remove();
+    });
+
+    it("returns a square factory that generates square based on symbolSize", () => {
+      let squareFactory = Plottable.SymbolFactories.square();
+      let actualSize = Math.pow(symbolSize, 2);
+      let d = squareFactory(symbolSize);
+      let expectedD = d3.svg.symbol().type("square").size(actualSize)(null);
+      assert.strictEqual(d, expectedD, "a square of set size is generated");
+      svg.append("path").attr("d", d);
+      let bbox = Plottable.Utils.DOM.elementBBox(svg.select("path"));
```

Comment:
Instead of using `svg.select("path")` you can store the above line in a variable and then pass the variable here